crates/rethnet_rpc_server/src/lib.rs

```diff
@@ -1 +1,200 @@
+                                                code: 0, message: String::from("No such account"), data: None,
+                                            }}
+                                        }
+                                        Err(e) => {
+                                            jsonrpc::ResponseData::<U256>::Error { error: jsonrpc::Error {
+                                                code: 0, message: e.to_string(), data: None,
+                                            }}
+                                        }
+                                    }
+                                }))
+                            }
+                            _ => { // TODO: after adding all the methods here, eliminate this
+                                   // catch-all match arm.
+                                Json(serde_json::json!(jsonrpc::Response {
+                                    jsonrpc: jsonrpc::Version::V2_0,
+                                    id,
+                                    data: jsonrpc::ResponseData::<serde_json::Value>::Error { error: jsonrpc::Error {
+                                        code: 0,
+                                        message: String::from("unsupported JSON-RPC method"),
+                                        data: Some(serde_json::to_value(method).unwrap()),
+                                    }}
+                                }))
+                            }
+                        }
+                    }
+                }
+            }
+        )
+    )
+    .with_state(state)
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use axum::{body::Body, http::Request};
+    use rethnet_eth::{
+        remote::{jsonrpc, BlockSpec, MethodInvocation},
+        Address,
+    };
+    use rethnet_evm::KECCAK_EMPTY;
+    use revm::primitives::state::AccountInfo;
+    use std::net::{SocketAddr, TcpListener};
+
+    fn start_server() -> SocketAddr {
+        let listener = TcpListener::bind("0.0.0.0:0".parse::<SocketAddr>().unwrap()).unwrap();
+        let address = listener.local_addr().unwrap();
+
+        let mut accounts: hashbrown::HashMap<Address, AccountInfo> = Default::default();
+        accounts.insert(
+            Address::from_low_u64_ne(1),
+            AccountInfo {
+                code: None,
+                code_hash: KECCAK_EMPTY,
+                balance: U256::ZERO,
+                nonce: 0,
+            },
+        );
+
+        let rethnet_state = Arc::new(Mutex::new(HybridState::with_accounts(accounts)));
+
+        tokio::spawn(async move {
+            axum::Server::from_tcp(listener)
+                .unwrap()
+                .serve(router(rethnet_state).await.into_make_service())
+                .await
+                .unwrap();
+        });
+
+        address
+    }
+
+    async fn submit_request(address: &SocketAddr, request: &RpcRequest) -> String {
+        String::from_utf8(
+            hyper::body::to_bytes(
```

Comment:
Why not use `reqwest`, which we already have as a dependency?

Reply:
i actually forgot about that. i just modeled this after the `axum` examples. but, if i understand correctly, this isn't introducing an additional dependency, because `hyper` is already a dependency of `axum`. do you think there's a good reason to pick one over the other for these tests?

Reply:
You do actually add the `hyper` dependency in the cargo toml. If not for this test, would you need it there?

Reply:
you're right, thanks for pushing. addressed in https://github.com/NomicFoundation/hardhat/pull/3998/commits/77ee710ca3d23139066c1ede9c62e0fe0091ccc8